### PR TITLE
Support for CalendarIntervalType and NullType

### PR DESCRIPTION
**Pull request by razajafri**
Handle nested CalendarIntervalType and NullType

Signed-off-by: Raza Jafri <rjafri@nvidia.com>

In addition to the changes to ParquetCachedBatchSerializer the following changes were made 
* Changed FuzzerUtils to handle nested types
* SPARK_CONF can be added on the command-line before running scala tests

### CODE REVIEW

**Comment by razajafri** [comment]
build

File: tests/src/test/scala/com/nvidia/spark/rapids/SparkQueryCompareTestSuite.scala [hunk 1]
```diff
@@ -79,7 +80,17 @@ object SparkSessionHolder extends Logging {
           "com.nvidia.spark.rapids.ExecutionPlanCaptureCallback")
         .config("spark.sql.warehouse.dir", sparkWarehouseDir.getAbsolutePath)
         .appName("rapids spark plugin integration tests (scala)")
-        .getOrCreate()
+
+    // comma separated config from command line
+    val commandLineVariables = System.getenv("SPARK_CONF")
```
Comment (razajafri) [inline tests/src/test/scala/com/nvidia/spark/rapids/SparkQueryCompareTestSuite.scala:85]: We can't change the static conf `spark.sql.cache.serializer`. This change allows for other options to be passed in to the tests like `spark.rapids.memory.gpu.allocFraction` etc

File: integration_tests/src/test/scala/com/nvidia/spark/rapids/tests/CacheTestSuite.scala [hunk 2]
```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
```
Comment (revans2) [inline integration_tests/src/test/scala/com/nvidia/spark/rapids/tests/CacheTestSuite.scala:2]: Was this written partly last year?

Comment (razajafri) [reply integration_tests/src/test/scala/com/nvidia/spark/rapids/tests/CacheTestSuite.scala]: I copied the license from a file that was. 

I have fixed the date. Please let me know if you have more comments

**Review by revans2 — APPROVED** [review]

**Comment by razajafri** [comment]
build

**Pull request merged**